java/rocksjni/sst_file_writerjni.cc

```diff
@@ -110,6 +110,43 @@ void Java_org_rocksdb_SstFileWriter_merge(JNIEnv *env, jobject jobj,
 
 /*
  * Class:     org_rocksdb_SstFileWriter
+ * Method:    merge
+ * Signature: (J[B[B)V
+ */
+void Java_org_rocksdb_SstFileWriter_merge__J_3B_3B(JNIEnv *env, jobject jobj,
+                                                   jlong jhandle, jbyteArray jkey,
+                                                   jbyteArray jval) {
+
+  jbyte* key = env->GetByteArrayElements(jkey, nullptr);
+  if(key == nullptr) {
+    // exception thrown: OutOfMemoryError
+    return;
+  }
+  rocksdb::Slice keySlice(
+      reinterpret_cast<char*>(key),  env->GetArrayLength(jkey));
```

Comment:
It seems like the array lengths are also passed in for some of the other native method implementations. For examples, see: https://github.com/facebook/rocksdb/blob/master/java/src/main/java/org/rocksdb/RocksDB.java#L2257-L2268 . 

I am not aware of the difference between passing in the array length explicitly vs. using env-> GetArrayLength() method used here and in line 134. Also, those functions (in the example I pointed out above) seem to be taking an 'offset' as well, but it is always set to 0. Not sure use of the usefulness of offset either. 

If a length and offset need to be passed in, the function signature needs to change to ...merge_J_3BII_3BII . 

@adamretter could you elaborate on the usefulness of passing in length and offset vs. using env->GetArrayLength()?